### PR TITLE
Make ShadyCSS types and externs stricter.

### PR DESCRIPTION
**Pull request by aomarks**
This makes ShadyCSS compile cleanly with some of the conformance checks for unresolved types, unknown this, and unknown properties from https://github.com/google/closure-compiler/wiki/JS-Conformance-Framework

### CODE REVIEW

File: src/style-util.js [hunk 1]
```diff
@@ -325,21 +325,21 @@ const CSS_BUILD_ATTR = 'css-build';
  * @return {string} Can be "", "shady", or "shadow"
  */
 export function getCssBuild(element) {
-  if (element['__cssBuild'] === undefined) {
```
Comment (TimvdLippe) [inline src/style-util.js]: This was a recent change to fix an undefined bug. Are we sure Closure now properly understands this?

Comment (aomarks) [reply src/style-util.js]: Yes, this PR also adds an extern for __cssBuild. Both the extern or the [] style access will cause Closure not to rename the property. I switched this to the externs style for consistency with all the other properties.

Comment (TimvdLippe) [reply src/style-util.js]: Ah I see. Thanks for the explanation!

**Comment by azakus** [comment]
I've updated WCT in #220, so I'll check manually.

**Comment by azakus** [comment]
LGTM with manual firefox testing

**Pull request merged**